### PR TITLE
[MAINTENANCE] Deprecate context.get_datasource

### PR DESCRIPTION
**Pull request by tyler-hoffman**
Deprecates `context.get_datasource`. `context.data_sources.get` should be used instead. Things to watch out for in here:
* I replaced the implementation of `get_datasource` so that it just calls into `context.data_sources.get`.
* These 2 methods raised different errors, and I'm keeping that behavior the same. But in some places where we were calling `get_datasource`, I had to update the surrounding try/except to now expect `KeyError`s
* I didn't see a test specifically around the success of our now-deprecated method, so I added one. Seems kinda weird, but it's deprecated; not removed.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -849,17 +851,11 @@ def get_datasource(self, name: str = "default") -> FluentDatasource:
         Raises:
             ValueError: The input `datasource_name` is None.
         """
-        if name is None:
-            raise ValueError("Must provide a datasource_name to retrieve an existing Datasource")  # noqa: TRY003
-
         try:
-            datasource = self.data_sources.all()[name]
+            return self.data_sources.get(name)
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py:860]: Could we emit a deprecation warning here?

Comment (cdkini) [reply great_expectations/data_context/data_context/abstract_data_context.py:860]: The decorator only updates how it presents in docs iirc

Comment (tyler-hoffman) [reply great_expectations/data_context/data_context/abstract_data_context.py:860]: Oh good call. I think the docstring for the decorator says it updates the docstring, but I guess that's only in the docs. Added a warning on there. Thanks!

Comment (cdkini) [reply great_expectations/data_context/data_context/abstract_data_context.py:860]: Could we add a `# deprecated-v1.2.0` comment above this to be consistent with other warnings?

Comment (tyler-hoffman) [reply great_expectations/data_context/data_context/abstract_data_context.py:860]: Done!

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 2]
```diff
@@ -849,17 +850,17 @@ def get_datasource(self, name: str = "default") -> FluentDatasource:
         Raises:
             ValueError: The input `datasource_name` is None.
         """
-        if name is None:
-            raise ValueError("Must provide a datasource_name to retrieve an existing Datasource")  # noqa: TRY003
-
+        # deprecated-v0.13.12
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py]: Pls change to current version :)

**Review by cdkini — APPROVED** [review]

**Pull request merged**